holoviews/plotting/bokeh/callbacks.py

```diff
@@ -350,78 +303,258 @@ def get_customjs(self, references):
             handle.callback = js_callback
 
 
 
+class ServerCallback(MessageCallback):
+    """
+    Implements methods to set up bokeh server callbacks. A ServerCallback
+    resolves the requested attributes on the Python end and then hands
+    the msg off to the general on_msg handler, which will update the
+    Stream(s) attached to the callback.
+    """
+
+    @classmethod
+    def resolve_attr_spec(cls, spec, cb_obj, model=None):
+        """
+        Resolves a Callback attribute specification looking the
+        corresponding attribute up on the cb_obj, which should be a
+        bokeh model. If not model is supplied cb_obj is assumed to
+        be the same as the model.
+        """
+        if not cb_obj:
+            raise Exception('Bokeh plot attribute %s could not be found' % spec)
+        if model is None:
+            model = cb_obj
+        spec = spec.split('.')
+        resolved = cb_obj
+        for p in spec[1:]:
+            if p == 'attributes':
+                continue
+            if isinstance(resolved, dict):
+                resolved = resolved.get(p)
+            else:
+                resolved = getattr(resolved, p, None)
+        return {'id': model.ref['id'], 'value': resolved}
+
+
+    def on_change(self, attr, old, new):
+        """
+        Process change events adding timeout to process multiple concerted
+        value change at once rather than firing off multiple plot updates.
+        """
+        self._queue.append((attr, old, new))
+        if self.process_on_change not in self.plot.document._session_callbacks:
+            self.plot.document.add_timeout_callback(self.process_on_change, 50)
+
+
+    def on_event(self, event):
+        """
+        Process bokeh UIEvents adding timeout to process multiple concerted
+        value change at once rather than firing off multiple plot updates.
+        """
+        self._queue.append((event))
+        if self.process_on_event not in self.plot.document._session_callbacks:
+            self.plot.document.add_timeout_callback(self.process_on_event, 50)
+
+
+    def process_on_event(self):
+        """
+        Trigger callback change event and triggering corresponding streams.
+        """
+        if not self._queue:
+            return
+        # Get unique event types in the queue
+        events = list(OrderedDict([(event.event_name, event)
+                                   for event in self._queue]).values())
+        self._queue = []
+
+        # Process event types
+        for event in events:
+            msg = {}
+            for attr, path in self.attributes.items():
+                model_obj = self.plot_handles.get(self.models[0])
+                msg[attr] = self.resolve_attr_spec(path, event, model_obj)
+            self.on_msg(msg)
+        self.plot.document.add_timeout_callback(self.process_on_event, 50)
+
+
+    def process_on_change(self):
+        if not self._queue:
+            return
+        self._queue = []
```

Comment:
Because process_on_change just looks up the model values directly, that avoids having separate callbacks for each model change, which would be required otherwise.